src/types.ts

```diff
@@ -41,14 +41,12 @@ export type BeforeLanguageSwitchHandler = <Context = unknown>(
   newLocale: string,
   initialSetup: boolean,
   context: Context
-) => string | void
+) => Promise<unknown>
 
-export type LanguageSwitchedHandler = (oldLocale: string, newLocale: string) => void
+export type LanguageSwitchedHandler = (oldLocale: string, newLocale: string) => Promise<unknown>
```

Comment:
This should also be removed.
Here is [same reason](https://github.com/nuxt-modules/i18n/pull/1919/files#r1133094629
LanguageSwitchedHandler) too.